src/core/net/dhcp6.hpp

```diff
@@ -91,68 +94,84 @@ typedef enum Type
     kTypeRelayReply         = 13,
     kTypeLeaseQuery         = 14,
     kTypeLeaseQueryReply    = 15,
-} Type;
+};
 
 /**
- * This class implements DHCPv6 header.
+ * This class represents a DHCP6 transaction identifier.
  *
  */
 OT_TOOL_PACKED_BEGIN
-class Dhcp6Header
+class TransactionId : public Equatable<TransactionId>, public Clearable<TransactionId>
 {
 public:
+    enum : uint16_t
+    {
+        kSize = 3, // Transaction Id size (in bytes).
+    };
+
     /**
-     * This method initializes the DHCPv6 header to all zeros.
+     * This method generates a cryptographically secure random sequence to populate the transaction identifier.
+     *
+     * @retval OT_ERROR_NONE     Successfully generated a random transaction identifier.
+     * @retval OT_ERROR_FAILED   Failed to generate random sequence.
      *
      */
-    void Init(void)
-    {
-        mType             = 0;
-        mTransactionId[0] = 0;
-    }
+    otError GenerateRandom(void) { return Random::Crypto::FillBuffer(m8, kSize); }
+
+private:
+    uint8_t m8[kSize];
+} OT_TOOL_PACKED_END;
 
+/**
+ * This class implements DHCPv6 header.
+ *
+ */
+OT_TOOL_PACKED_BEGIN
+class Header : public Clearable<Header>
+{
+public:
     /**
      * This method returns the DHCPv6 message type.
      *
      * @returns The DHCPv6 message type.
      *
      */
-    Type GetType(void) const { return static_cast<Type>(mType); }
+    Type GetType(void) const { return mType; }
 
     /**
      * This method sets the DHCPv6 message type.
      *
      * @param[in]  aType  The DHCPv6 message type.
      *
      */
-    void SetType(Type aType) { mType = static_cast<uint8_t>(aType); }
+    void SetType(Type aType) { mType = aType; }
 
     /**
-     * This method returns the DHCPv6 message transaction id.
+     * This method returns the DHCPv6 message transaction aTransaction.
      *
-     * @returns A pointer of DHCPv6 message transaction id.
+     * @returns The DHCPv6 message transaction identifier.
      *
      */
-    uint8_t *GetTransactionId(void) { return mTransactionId; }
+    const TransactionId &GetTransactionId(void) { return mTransactionId; }
 
     /**
-     * This method sets the DHCPv6 message transaction id.
+     * This method sets the DHCPv6 message transaction identifier.
      *
-     * @param[in]  aBuf  The DHCPv6 message transaction id.
+     * @param[in]  aTransactionId  The DHCPv6 message transaction identifier.
      *
      */
-    void SetTransactionId(uint8_t *aBuf) { memcpy(mTransactionId, aBuf, kTransactionIdSize); }
+    void SetTransactionId(const TransactionId &aTransactionId) { mTransactionId = aTransactionId; }
 
 private:
-    uint8_t mType;                              ///< Type
-    uint8_t mTransactionId[kTransactionIdSize]; ///< Transaction Id
+    Type          mType;
+    TransactionId mTransactionId;
 } OT_TOOL_PACKED_END;
 
 /**
- * DHCPv6 Option Codes
+ * DHCPv6 Option Codes.
  *
  */
-typedef enum Code
+enum Code : uint8_t
```

Comment:
@jwhui I had this originally as `uint16_t` type as well. Noticed that there was some unexpected code-size increase, after hunting it down, realized (to my surprise) that it was being caused by using `uint16_t` instead of `uint8` as the `enum` underlying int type. 

Here is the diff with just changing `Code` to be `uint16_t` instead of `uint8_t` (+64 bytes):
```
   text   data    bss    dec    hex filename
 390808    736  72084 463628  7130c output/nrf52840/bin/ot-ncp-ftd  (with enum Code : uint8_t)
 390872    736  72084 463692  7134c output/nrf52840/bin/ot-ncp-ftd  (with enum Code : uint16_t)
```
I opted for using `uint8_t` (after checking that all values are indeed less than 255) instead. We  can consider the `enum Code` as our representation of DHCP6 Option Codes and a subset of values that we use and care to know. In the `Option` itself we need to `HostSwa1p16()` the value and cast it to `Code`. 

Interestingly for the `StatusCode` and `DuidType` changing from `uint8_t` to `uint16_t` does not seem to imapct code size. So I will change them.

What do you think?

Reply:
Actually thinking more on this, I would say it's safer (and more correct) to declare it as `uint16_t` sicne in `Option` class we are reading it as `uint16_t` then doing `HostSwap16()` and casting it as `Code`. We don't want to loose the read value range here otherwise we may read the incorrect Code value.

Changed it to new push (new small commit to be squashed).